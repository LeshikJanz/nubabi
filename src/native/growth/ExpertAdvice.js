// @flow
import type { Avatar } from '../../common/types';
import React from 'react';
import { Image } from 'react-native';
import { gql } from 'react-apollo';
import { Box, Text } from '../components';

type Props = {
  avatar: Avatar,
  name: string,
  discipline: string,
};

export const ExpertAdvice = ({ name, discipline, avatar }: Props) => {
  return (
    <Box borderBottomWidth={1} style={() => ({ borderColor: '#E9ECF4' })}>
      <Box padding={1}>
        <Text
          bold
          style={theme => ({ color: theme.colors.open.blue0 })}
          marginBottom={1}
        >
          EXPERT ADVICE BY
        </Text>
        <Box flex={1} flexDirection="row" justifyContent="center">
          <Image
            source={{ uri: avatar.url }}
            style={{ width: 46, height: 45 }}
            borderRadius={46 / 2}
          />
          <Box flex={1} justifyContent="space-around" marginLeft={1}>
            <Text bold spacing={-0.39} size={1}>
              {name}
            </Text>
            <Text color="secondary" spacing={-0.39}>
              {discipline}
            </Text>
          </Box>
        </Box>
      </Box>
    </Box>
  );
};

ExpertAdvice.fragments = {
  expert: gql`
    fragment ExpertAdvice on Expert {
      name
      discipline
      avatar {
        url
      }
    }
  `,
};

export default ExpertAdvice;
