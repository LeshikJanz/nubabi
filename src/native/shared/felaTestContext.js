import React from 'react';
import { NativeFelaProvider } from '../components/FelaProvider';
import { createRenderer } from 'fela-native';
import theme from '../../common/themes/defaultTheme';

export const renderer = createRenderer();
const FelaProvider = NativeFelaProvider(renderer, theme);

const felaTestContext = component =>
  <FelaProvider>
    {component}
  </FelaProvider>;

export default felaTestContext;
