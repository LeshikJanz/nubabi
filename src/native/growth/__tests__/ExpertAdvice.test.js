import 'react-native';
import React from 'react';
import { ExpertAdvice } from '../ExpertAdvice';
import renderer from 'react-test-renderer';
import felaTestContext from '../../shared/felaTestContext';

test('it renders correctly', () => {
  const component = (
    <ExpertAdvice
      name="Test Expert"
      discipline="Test Doctor"
      avatar={{ url: 'http://example.com/person.png' }}
    />
  );

  const tree = renderer.create(felaTestContext(component)).toJSON();

  expect(tree).toMatchSnapshot();
});
