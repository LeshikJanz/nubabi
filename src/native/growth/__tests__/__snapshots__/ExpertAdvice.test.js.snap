// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "paddingBottom": 16,
    "paddingLeft": 16,
    "paddingRight": 16,
    "paddingTop": 16,
    "marginBottom": 16,
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#748294",
    "fontFamily": "SF Pro Text",
    "fontWeight": "600",
    "flex": 1,
    "flexDirection": "row",
    "justifyContent": "space-around",
    "marginLeft": 16
}

<View
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "#E9ECF4",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "color": "#33B7EB",
          "fontFamily": "SF Pro Text",
          "fontSize": 13,
          "fontWeight": "600",
          "lineHeight": 16,
          "marginBottom": 16,
        }
      }
    >
      EXPERT ADVICE BY
    </Text>
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        borderRadius={23}
        source={
          Object {
            "uri": "http://example.com/person.png",
          }
        }
        style={
          Object {
            "height": 45,
            "width": 46,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#9EABBC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "flex": 1,
            "justifyContent": "space-around",
            "marginLeft": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          spacing={-0.39}
          style={
            Object {
              "borderBottomWidth": 0,
              "borderColor": "#9EABBC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "color": "#454D56",
              "fontFamily": "SF Pro Text",
              "fontSize": 13.866666666666667,
              "fontWeight": "600",
              "lineHeight": 16,
            }
          }
        >
          Test Expert
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          spacing={-0.39}
          style={
            Object {
              "borderBottomWidth": 0,
              "borderColor": "#9EABBC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "color": "#748294",
              "fontFamily": "SF Pro Text",
              "fontSize": 13,
              "lineHeight": 16,
            }
          }
        >
          Test Doctor
        </Text>
      </View>
    </View>
  </View>
</View>
`;
